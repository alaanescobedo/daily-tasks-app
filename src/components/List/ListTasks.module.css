.container {
    display: flex;
    flex-direction: column;
    gap: 1rem;
    width: 95%;
    margin: 0 auto;
}