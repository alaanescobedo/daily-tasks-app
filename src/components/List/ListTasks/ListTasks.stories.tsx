import { ComponentStory, ComponentMeta } from '@storybook/react'
import { tasks } from '../../../config/test-tasks'

import { ListTasks as ListTasksComponent, ListTasksProps } from './ListTasks'

const ListTasksStory: ComponentMeta<typeof ListTasksComponent> = {
  title: 'List/List Of Tasks',
  component: ListTasksComponent
}
export default ListTasksStory

const Template: ComponentStory<typeof ListTasksComponent> = (args) => <ListTasksComponent {...args} />

export const ListOfTasks = Template.bind({})
const listTasksArgs: ListTasksProps = {
  tasks
}
ListOfTasks.args = listTasksArgs
