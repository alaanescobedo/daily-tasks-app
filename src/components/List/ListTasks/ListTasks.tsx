import { Task } from '@hooks'
import { TaskCard } from '@components/Card'

import styles from './ListTasks.module.css'

export interface ListTasksProps {
  tasks: { [key: string]: Task[] }
}

export const ListTasks = ({ tasks }: ListTasksProps): JSX.Element => {
  return (
    <ul className={styles.container}>
      {Object.entries(tasks).map(([day, tasks]: any) => (
        <TaskCard tasks={tasks} day={day} key={day} navigateTo={`/activities/${day as string}`} />
      ))}
    </ul>
  )
}
