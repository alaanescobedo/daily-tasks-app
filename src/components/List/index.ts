export * from './ListActivities'
export * from './ListTasks'
