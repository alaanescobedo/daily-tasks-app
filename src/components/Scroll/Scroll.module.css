.container {
    display: flex;
    flex-direction: column;
    overflow: scroll;
    height: 65vh;
    border-radius: 10px;
}