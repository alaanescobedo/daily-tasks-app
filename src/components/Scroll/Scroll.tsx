import styles from './Scroll.module.css'

export interface ScrollProps {
  children: React.ReactNode
}

export const Scroll = ({ children }: ScrollProps): JSX.Element => {
  return (
    <div className={styles.container}>
      {children}
    </div>
  )
}
