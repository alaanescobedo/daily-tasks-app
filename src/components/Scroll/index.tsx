import { TaskCard } from '../Card'
import styles from './Scroll.module.css'

export const Tasks = (): JSX.Element => {
  return (
    <div className={styles.container}>
      <TaskCard />
      <TaskCard />
      <TaskCard />
      <TaskCard />
      <TaskCard />
      <TaskCard />
      <TaskCard />
      <TaskCard />
    </div>
  )
}
