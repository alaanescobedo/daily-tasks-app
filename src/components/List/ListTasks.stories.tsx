import { ComponentStory, ComponentMeta } from '@storybook/react'
import { Task } from '../../hooks/useTasks'

import { ListTasks as ListTasksComponent, ListTasksProps } from './ListTasks'

const ListTasksStory: ComponentMeta<typeof ListTasksComponent> = {
  title: 'Inputs/List',
  component: ListTasksComponent
}
export default ListTasksStory

const Template: ComponentStory<typeof ListTasksComponent> = (args) => <ListTasksComponent {...args} />
const tasks: { [key: string]: Task[] } = {
  '2022-04-11': [
    {
      title: 'Task 1',
      createdAt: '2020-01-01',
      entityId: Math.random().toString(),
      scheduledFor: new Date(Date.now() + 3 * 1000 * 60 * 60 * 24).toISOString(),
      status: 'Pending',
      userID: 'user123'
    }
  ],
  '2022-04-12': [
    {
      title: 'Task 2',
      createdAt: '2020-01-01',
      entityId: Math.random().toString(),
      scheduledFor: new Date(Date.now() + 1 * 1000 * 60 * 60 * 24).toISOString(),
      status: 'Pending',
      userID: 'user123'
    }, {
      title: 'Task 3',
      createdAt: '2020-01-01',
      entityId: Math.random().toString(),
      scheduledFor: new Date(Date.now() + 1 * 1000 * 60 * 60 * 24).toISOString(),
      status: 'Pending',
      userID: 'user123'
    }
  ]
}

export const ListTasks = Template.bind({})
const listTasksArgs: ListTasksProps = {
  tasks
}
ListTasks.args = listTasksArgs
