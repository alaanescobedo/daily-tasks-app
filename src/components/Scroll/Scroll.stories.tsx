import { ComponentStory, ComponentMeta } from '@storybook/react'
import { ListTasksProps } from '@components/List/ListTasks'
import { ListOfTasks } from '@components/List/ListTasks/ListTasks.stories'

import { Scroll as ScrollComponent, ScrollProps } from './Scroll'

const ScrollStory: ComponentMeta<typeof ScrollComponent> = {
  title: 'Scroll/Scroll',
  component: ScrollComponent
}
export default ScrollStory

const Template: ComponentStory<typeof ScrollComponent> = (args) => <ScrollComponent {...args} />

export const Scroll = Template.bind({})
const scrollProps: ScrollProps = {
  children: <ListOfTasks {...ListOfTasks.args as ListTasksProps} />
}
Scroll.args = scrollProps
