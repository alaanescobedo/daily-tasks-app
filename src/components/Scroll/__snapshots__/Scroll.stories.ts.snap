// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Scroll/Scroll Scroll 1`] = `
Array [
  " ",
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <span
      className="app_container"
    >
       
      <div
        className="container"
      >
        <ul
          className="container"
        >
          <a
            className=""
            href="/activities/2022-04-11"
            onClick={[Function]}
          >
            <div
              className="container"
            >
              <ul
                className="list"
              >
                <li>
                  - 
                  Task 1
                </li>
              </ul>
              <div
                className="day"
              >
                <p>
                  2022-04-11
                </p>
              </div>
            </div>
          </a>
          <a
            className=""
            href="/activities/2022-04-12"
            onClick={[Function]}
          >
            <div
              className="container"
            >
              <ul
                className="list"
              >
                <li>
                  - 
                  Task 2
                </li>
                <li>
                  - 
                  Task 3
                </li>
              </ul>
              <div
                className="day"
              >
                <p>
                  2022-04-12
                </p>
              </div>
            </div>
          </a>
          <a
            aria-current="page"
            className="active"
            href="/activities/2022-04-13"
            onClick={[Function]}
          >
            <div
              className="container"
            >
              <ul
                className="list"
              >
                <li>
                  - 
                  Task 4
                </li>
              </ul>
              <div
                className="day"
              >
                <p>
                  2022-04-13
                </p>
              </div>
            </div>
          </a>
          <a
            className=""
            href="/activities/2022-04-14"
            onClick={[Function]}
          >
            <div
              className="container"
            >
              <ul
                className="list"
              >
                <li>
                  - 
                  Task 5
                </li>
              </ul>
              <div
                className="day"
              >
                <p>
                  2022-04-14
                </p>
              </div>
            </div>
          </a>
          <a
            className=""
            href="/activities/2022-04-15"
            onClick={[Function]}
          >
            <div
              className="container"
            >
              <ul
                className="list"
              >
                <li>
                  - 
                  Task 6
                </li>
              </ul>
              <div
                className="day"
              >
                <p>
                  2022-04-15
                </p>
              </div>
            </div>
          </a>
        </ul>
      </div>
       
    </span>
     
  </div>,
  " ",
  " ",
]
`;
